stix2/markings/__init__.py

```diff
@@ -0,0 +1,214 @@
+"""
+Python STIX 2.0 Data Markings API.
+
+These high level functions will operate on both object level markings and
+granular markings unless otherwise noted in each of the functions.
+"""
+
+from stix2.markings import granular_markings, object_markings
+
+
+def get_markings(obj, selectors, inherited=False, descendants=False):
+    """
+    Get all markings associated to the field(s).
+
+    Args:
+        obj: An SDO or SRO object.
+        selectors: string or list of selectors strings relative to the SDO or
+            SRO in which the properties appear.
+        inherited: If True, include object level markings and granular markings
+            inherited relative to the properties.
+        descendants: If True, include granular markings applied to any children
+            relative to the properties.
+
+    Returns:
+        list: Marking identifiers that matched the selectors expression.
+
+    Note:
+        If ``selectors`` is None, operation will be performed only on object
+        level markings.
+
+    """
+    if selectors is None:
+        return object_markings.get_markings(obj)
+
+    results = granular_markings.get_markings(
+        obj,
+        selectors,
+        inherited,
+        descendants
+    )
+
+    if inherited:
+        results.extend(object_markings.get_markings(obj))
+
+    return list(set(results))
+
+
+def set_markings(obj, selectors, marking):
```

Comment:
`selectors` should be kwarg defaulting to `None`. That means the order of args will have to change, which I'm not sure that I like, as it makes the function signatures somewhat inconsistent.

Reply:
Is it more important to have consistent signatures or to be pythonic? We could make `marking` a kwarg with a default of `None` as well, and then check and throw an error if `marking` is `None`.

Reply:
It's actually more consistent to have `markings` come before `selectors` (if you compare across all three files, especially `object_markings.py`) on functions that need to accept markings (not `get`, `clear`, or `is_marked`). I was too focused on making selectors the first argument, rather than ensuring consistent arguments for the same function across all modules. 

Here's what I'm thinking:

`markings/__init__.py`
```python
def get_markings(obj, selectors=None, inherited=False, descendants=False):
def set_markings(obj, marking, selectors=None):
def remove_markings(obj, marking, selectors=None):
def add_markings(obj, marking, selectors=none):
def clear_markings(obj, selectors=None):
def is_marked(obj, marking=None, selectors=None, inherited=False, descendants=False):
```

`markings/object_markings.py`
```python
def get_markings(obj):
def add_markings(obj, marking):
def remove_markings(obj, marking):
def set_markings(obj, marking):
def clear_markings(obj):
def is_marked(obj, marking=None):
```

`markings/granular_markings.py`
```python
def get_markings(obj, selectors, inherited=False, descendants=False):
def set_markings(obj, marking, selectors):
def remove_markings(obj, marking, selectors):
def add_markings(obj, marking, selectors):
def clear_markings(obj, selectors):
def is_marked(obj, marking=None, selectors, inherited=False, descendants=False):
```

The last one is tricky, because it obviously isn't valid. `selectors` is 'required' for `granular_markings.is_marked`, but would it make sense to give it a default value (in all of the `granular_markings` functions, of `[""]`?

Reply:
I like this. For `granular_markings.is_marked()`, I think we can assume that if users don't provide a value for `selectors` then it is not what they intended to do and should raise an error. If they don't provide a selector to `markings.is_marked` it will call the `object_markings` version, so if this function is called and no selector is provided, they are calling the `granular_markings` version directly and probably just forgot to add that arg. The same is true for the other `granular_markings` functions, but by defining them as required args, Python will raise those errors. For `is_marked`, I think we should keep the argument order consistent, add a check, and raise the error ourselves.

I don't think `[""]` has any advantage over `None` as a default, and `None` will be easier for code maintainers to understand its purpose.